public/scripts/api-helpers.js

```diff
@@ -131,4 +117,71 @@ const callAPIByCategory = async (task) => {
     default:
       return task;
   }
-}
+};
+
+/**
+ * Gives a time limit to a promise
+ * Given promise is rejected if not resolved before timeout
+ * @param  {number}      ms      Length of ms before timeout
+ * @param  {Promise<{}>} promise Promise to limit
+ * @return {Promise<>}
+ */
+const timeoutPromise = (ms, promise) => {
+  const timeout = new Promise((resolve, reject) =>
+    setTimeout(() =>
+      reject(`Timed out after ${ms} ms.`), ms));
+
+  return Promise.race([
+    promise,
+    timeout
+  ]);
+};
+
+/**
+ * Checks if given word is found in the given string
+ * @param  {string}  str  String to check condition
+ * @param  {string}  word Word to look for in the string
+ * @return {boolean}      Boolean if word found in string
+ */
+const findWordInString = (str, word) => {
+  return new RegExp('\\b' + word + '\\b', 'i').test(str);
+};
+
+/**
+ * Matches string against key values to check for easy API match
+ * @param  {string} string Takes in string to check for keyword
+ * @return {string}        String of category matched, else undefined
+ */
+const matchCategoryKeyword = (string) => {
+  let category;
+  const categories = ['restaurants', 'films', 'books', 'products'];
+  const keywords = ['eat', 'watch', 'read', 'buy'];
+
+  keywords.some((val, index) => {
+    if (findWordInString(string, val)) {
+      category = categories[index];
+    }
+  });
+
+  return category;
+};
+
+/**
+ * Trim given string of matched keyword and whitespace from the start/end
+ * @param  {string} string Takes in string to check for keyword
+ * @return {string}        Trimmed string
+ */
+const filterKeywords = (string) => {
+  const keywords = ['eat', 'watch', 'read', 'buy'];
+  let newString = '';
+
+  keywords.some((val, index) => {
```

Comment:
Same problem here. 
It filters out all keywords so movie titles like "The Reader" and "Eat, Love, Pray" would be altered

Reply:
The Reader actually wouldn't be affected, because of the `\\b` word boundaries. Though it would change for 'watch eat, pray, love' or something like 'buy an apple watch'. So still something to consider, ya.

Reply:
filterKeyowrds wont filter anything for me, i just tried

Reply:
The reason is the replace statement is case sensitive. Need to change it to regExp replace with "i"